src/Feature/Editor/EditorSurface.re

```diff
@@ -356,641 +166,97 @@ let%component make =
-               )
-             });
-
-          /* Draw background for line numbers */
-          if (showLineNumbers != LineNumber.Off) {
-            Skia.Paint.setColor(
-              rectPaint,
-              theme.editorLineNumberBackground |> Color.toSkia,
-            );
-            CanvasContext.drawRectLtwh(
-              ~left=0.,
-              ~top=0.,
-              ~width=lineNumberWidth,
-              ~height=float(height),
-              ~paint=rectPaint,
-              canvasContext,
-            );
-
-            lineNumberPaint
-            |> Option.iter(paint => {
-                 ImmediateList.render(
-                   ~scrollY,
-                   ~rowHeight,
-                   ~height=float(height),
-                   ~count,
-                   ~render=
-                     (item, offset) => {
-                       renderLineNumber(
-                         paint,
-                         descenderHeight,
-                         fontWidth,
-                         fontHeight,
-                         item,
-                         lineNumberWidth,
-                         theme,
-                         showLineNumbers,
-                         cursorLine,
-                         offset,
-                         canvasContext,
-                       )
-                     },
-                   (),
-                 )
-               });
-          };
-
-          Option.iter(
-            EditorDiffMarkers.render(
-              ~scrollY,
-              ~rowHeight,
-              ~x=lineNumberWidth,
-              ~height=float(height),
-              ~width=Constants.diffMarkerWidth,
-              ~count,
-              ~canvasContext,
-              ~theme,
-            ),
-            diffMarkers,
-          );
-
-          if (shouldRenderIndentGuides) {
-            switch (activeBuffer) {
-            | None => ()
-            | Some(buffer) =>
-              IndentLineRenderer.render(
-                ~canvasContext,
-                ~buffer,
-                ~startLine=topVisibleLine - 1,
-                ~endLine=bottomVisibleLine + 1,
-                ~lineHeight=fontHeight,
-                ~fontWidth,
-                ~cursorLine=Index.toZeroBased(cursorPosition.line),
-                ~theme,
-                ~indentationSettings=indentation,
-                ~bufferPositionToPixel,
-                ~showActive=shouldHighlightActiveIndentGuides,
-                (),
-              )
-            };
-          };
-        }}
-      />
-      <Opacity opacity=cursorOpacity> <View style=cursorStyle /> </Opacity>
-      <View style=horizontalScrollBarStyle>
-        <EditorHorizontalScrollbar
-          editor
-          metrics
-          width={int_of_float(layout.bufferWidthInPixels)}
-          theme
-        />
-      </View>
-    </View>
-    minimapLayout
-    hoverElements
-    <View style=verticalScrollBarStyle>
+  let (gutterWidth, gutterView) =
+    <GutterView
```

Comment:
Thanks for factoring this `<GutterView />` out! Much cleaner to have it separate. I think this can actually help us chip away at the `EditorLayout` too.